File: Programming Basics/While Loop - Exercises/02.ExamPreparation.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static void Main(string[] args)
    {
        int unsatisfiedGrades = int.Parse(Console.ReadLine());
        string taskName = Console.ReadLine();
        int grade = int.Parse(Console.ReadLine());
        List<double> grades = new List<double>();

        int poorGradesCount = 0;
        int taskCount = 0;
        string lastTask = null;

        while (taskName != "Enough")
        {
            taskCount++;
            if (grade <= 4)
            {
                poorGradesCount++;
                if (poorGradesCount == unsatisfiedGrades)
                {
                    Console.WriteLine($"You need a break, {poorGradesCount} poor grades.");
                    System.Environment.Exit(0);
                }
            }
            grades.Add(grade);

            lastTask = taskName;
            taskName = Console.ReadLine();
            if (taskName == "Enough")
                break;
            grade = int.Parse(Console.ReadLine());
        }
        Console.WriteLine($"Average score: {grades.Average():f2}");
        Console.WriteLine($"Number of problems: {taskCount}");
        Console.WriteLine($"Last problem: {lastTask}");
    }
}

File: Programming Basics - Python/While Loop - Exercises/02.ExamPreparation.py
number_unsatisfying_marks = int(input())

task = input()

total_score = 0
number_of_tasks = 0
unsatisfying_marks = 0

last_problem = None

while task != "Enough":
    score = int(input())

    last_problem = task
    total_score += score
    number_of_tasks += 1

    if score <= 4:
        unsatisfying_marks += 1
        if number_unsatisfying_marks == unsatisfying_marks:
            print(f"You need a break, {unsatisfying_marks} poor grades.")
            exit()

    task = input()

print(f"Average score: {total_score / number_of_tasks:.2f}")
print(f"Number of problems: {number_of_tasks}")
print(f"Last problem: {last_problem}")